core/blockchain.go

```diff
@@ -551,7 +552,7 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis
 }
 
 // NewParallelBlockChain , similar to NewBlockChain, creates a new blockchain object, but with a parallel state processor
-func NewParallelBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis, overrides *ChainOverrides, engine consensus.Engine, vmConfig vm.Config, shouldPreserve func(header *types.Header) bool, txLookupLimit *uint64, checker ethereum.ChainValidator, numprocs int) (*BlockChain, error) {
+func NewParallelBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis, overrides *ChainOverrides, engine consensus.Engine, vmConfig vm.Config, shouldPreserve func(header *types.Header) bool, txLookupLimit *uint64, checker ethereum.ChainValidator, numprocs int, enforce bool) (*BlockChain, error) {
```

Comment:
Could you please add a unit test for testing when `enforce=true`?

Reply:
Thanks, done in this [commit](https://github.com/maticnetwork/bor/pull/1425/commits/b9ff8c0e90eb77fab31bb59d2472c2f2060efb95). 

Note that lot of other tests use `newCanonical` function to generate a chain (this is unrelated to changes in this PR). That function still uses serial execution. The change I made only enforces block-stm in a another call later (`testBlockChainImport`). If we want the former function to also use block-stm, we should explore that maybe in another task.